### PR TITLE
fix: works when Gitlab runner is not at the same machine as Gitlab

### PR DESCRIPTION
**Pull request by softaria-roman**
Hello,

Thank you for this great product.
This little patch fixes the following issue with Gitlab integration:
Currently, it uses localhost to communicate with GitLab via API.
While in most real-life installations, gitlab runners and gitlab itself run on different machines,
So, localhost does not work.
Fortunately, gitlab has a special environment variable with its url.
The patch uses it. The patch was tested in a real environment.

### CODE REVIEW

File: code-review-gpt/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "lint-fix": "eslint . --ext=js,ts,tsx --fix",
     "lint-test": "eslint . --ext=js,ts",
     "start": "ts-node ./src/index.ts",
-    "review": "ts-node ./src/index.ts review",
+    "review": "ts-node ./src/index.ts review --model=gpt-3.5-turbo --debug=true",
```
Comment (mattzcarey) [inline code-review-gpt/package.json:14]: please remove this

File: code-review-gpt/src/common/ci/gitlab/commentOnPR.ts [hunk 2]
```diff
@@ -1,7 +1,192 @@
-import { Gitlab } from "@gitbeaker/rest";
+import {
+  DiffRefsSchema,
+  DiscussionNotePositionOptions,
+  Gitlab,
+} from "@gitbeaker/rest";
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (mattzcarey) [inline code-review-gpt/src/common/ci/gitlab/commentOnPR.ts:6]: why do you need this?

**Comment by mattzcarey** [comment]
Hey thanks very much for your contribution. I love the better GitLab implementation. Unfortunately, I can't merge it as is. 

- Please can you clean up the readmes? 
- If you no longer want to show the emoji summary can you add it as a flag to switch it off? 
- Similar to showing all reviews. This should be a flag. Imagine changing 30 files. It will become unreadable.
- Removing the decoding of the answer means that whenever there is a code block suggested, that feedback will fail. This is not the desired functionality.

Many thanks, Matt

**Comment by softaria-roman** [comment]
Sorry. 
I intended to send only the first commit (which fixes one bug) not all 3 (with a lot of changes).
Please close this PR and I will send new with the only required things.

I guess I will send 2 PRs - one with the bug fixes and another one with the gitlab staff.

I will not touch emoji and other things.

Will it be okay?

**Comment by mattzcarey** [comment]
I still cant merge this. Please can you check the diff.

**Comment by borjafv** [comment]
Any news about this? It would be great to have a better Gitlab integration. Thanks!

**Comment by softaria-roman** [comment]
> Any news about this? It would be great to have a better Gitlab integration. Thanks!

Sorry, I am really too busy now :(
I keep it in mind.

Also, feel free to just get any code you find useful

**Pull request closed without merging**